File: e2e/selenide-junit5-selenium-grid/src/test/java/provider/StepProvider.java
package provider;

import interaction.step.*;
import lombok.Getter;

@Getter
public class StepProvider {
    public TaskTableStep tasks = new TaskTableStep();
    public LanguageSwitcherStep language = new LanguageSwitcherStep();
    public NavigationStep navigation = new NavigationStep();
    public AccessibilityStep accessibility = new AccessibilityStep();
}

File: demo-android/app/src/androidTest/java/com/example/demo/e2e/provider/StepProvider.kt
package com.example.demo.e2e.provider

import androidx.compose.ui.test.junit4.AndroidComposeTestRule
import com.example.demo.MainActivity
import com.example.demo.e2e.interaction.step.AccessibilityStep
import com.example.demo.e2e.interaction.step.CreateTaskStep
import com.example.demo.e2e.interaction.step.EditTaskStep
import com.example.demo.e2e.interaction.step.LanguageSwitcherStep
import com.example.demo.e2e.interaction.step.NavigationStep
import com.example.demo.e2e.interaction.step.TaskTableStep

class StepProvider(
    rule: AndroidComposeTestRule<*, MainActivity>,
) {
    val tasks: TaskTableStep = TaskTableStep(rule)
    val language: LanguageSwitcherStep = LanguageSwitcherStep(rule)
    val navigation: NavigationStep = NavigationStep(rule)
    val accessibility: AccessibilityStep = AccessibilityStep(rule)
}
